thesis-api-v2/src/main/java/pl/edu/agh/quizzesthesis/business/service/DefinitionProcessingService.java

```diff
@@ -174,6 +174,24 @@ public DefinitionProcessing removeAnswerFromDefinition() {
             return new DefinitionProcessing(newDefinition, answer, articleTitle);
         }
 
+        public DefinitionProcessing removeInBracketsFromDefinition() {
+            String text = definition;
+            if (text.length() > 1 && text.charAt(text.length() - 2) == '.') {
+                text = text.substring(0, text.length() - 1);
+            }
+            int startIndex = text.indexOf("(");
+            int endIndex = text.indexOf(")");
+            while (startIndex != -1 && endIndex != -1) {
```

Comment:
What about double bracket situation? For example `Some text ((Some text))` will return single `Some text )` as a result.

Reply:
I fixed it with code like this
`
            String text = definition;
            int startIndex = text.indexOf("(");
            int endIndex = text.indexOf(")");
            while (startIndex != -1 && endIndex != -1) {
                int potentialStartIndex = text.indexOf("(", startIndex + 1);
                while (potentialStartIndex != -1 && potentialStartIndex < endIndex) {
                    endIndex = text.indexOf(")", endIndex + 1);
                    potentialStartIndex = text.indexOf("(", potentialStartIndex + 1);
                }
                if (endIndex == -1)
                    endIndex = text.length();
                String replacement = "";
                String toBeReplaced = text.substring(startIndex, endIndex + 1);
                text = text.replace(toBeReplaced, replacement);
                startIndex = text.indexOf("(");
                endIndex = text.indexOf(")");
            }
`

Reply:
Just from reading it, it looks good. I don't have much time to test it so I will just believe you that you tested it and it works.